src/Microsoft.Extensions.FileSystemGlobbing/MatcherExtensions.cs

```diff
@@ -40,5 +40,28 @@ public static IEnumerable<string> GetResultsInFullPath(this Matcher matcher, str
 
             return result;
         }
+
+        /// <summary>
+        /// Matches the files passed in with the patterns in the matcher without going to disk.
+        /// </summary>
+        /// <param name="matcher">The matcher that holds the patterns and pattern matching type.</param>
+        /// <param name="files">The files to run the matcher against.</param>
+        /// <returns>The match results.</returns>
+        public static PatternMatchingResult Match(this Matcher matcher, IEnumerable<string> files)
+        {
+            return Match(matcher, Directory.GetCurrentDirectory(), files);
+        }
+
+        /// <summary>
+        /// Matches the files passed in with the patterns in the matcher without going to disk.
+        /// </summary>
+        /// <param name="matcher">The matcher that holds the patterns and pattern matching type.</param>
+        /// <param name="rootDir">The root directory for the matcher to match the files from.</param>
+        /// <param name="files">The files to run the matcher against.</param>
+        /// <returns>The match results.</returns>
+        public static PatternMatchingResult Match(this Matcher matcher, string rootDir, IEnumerable<string> files)
+        {
+            return matcher.Execute(new InMemoryDirectoryInfo(rootDir, files));
```

Comment:
Yeah it will :(
Any suggested workaround besides exposing the InMemoryDirectoryInfo